semantics/src/main/scala/aqua/semantics/header/ModuleSem.scala

```diff
@@ -30,16 +31,18 @@ class ModuleSem[S[_]: Comonad, C: Picker](expr: ModuleExpr[S])(using
         name.value,
         shouldDeclare
       ),
-      (ctx, _) =>
+      (ctx, initCtx) =>
         // When file is handled, check that all the declarations exists
         if (declareAll.nonEmpty)
           ctx.setModule(name.value, declares = ctx.all).validNec
-        else
+        else {
+          // summarize contexts to allow redeclaration of imports
+          val sumCtx = ctx |+| initCtx
```

Comment:
I guess it is more reasonable to do `initCtx |+| ctx`? Operation is not commutative and logically `initCtx` goes before `ctx`. WDYT?